ClientCore/ClientConfiguration.cs

```diff
@@ -382,15 +381,53 @@ public IEnumerable<string> SupplementalMapFileExtensions
 
         public OSVersion GetOperatingSystemVersion()
         {
+#if NETFRAMEWORK
+            // OperatingSystem.IsWindowsVersionAtLeast() is the preferred API but is not supported on earlier .NET versions
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                Version osVersion = Environment.OSVersion.Version;
+
+                if (osVersion.Major <= 4)
+                    return OSVersion.UNKNOWN;
+
+                if (osVersion.Major == 5)
+                    return OSVersion.WINXP;
+
+                if (osVersion.Major == 6 && osVersion.Minor == 0)
+                    return OSVersion.WINVISTA;
+
+                if (osVersion.Major == 6 && osVersion.Minor <= 1)
+                    return OSVersion.WIN7;
```

Comment:
Best OS

Reply:
About Windows versions, what about Windows 11? Is it included in WIN810? If yes, IMO the name should be updated to WIN81011 (or something more visually reasonable).

Reply:
`WIN810` includes Windows 11.

This question is not related to the .NET 4.8/8.0 so I'll just mark it as solved. Could write an issue about this.